src/tests/ftest/control/version.py

```diff
@@ -22,32 +22,64 @@ class DAOSVersion(TestWithServers):
     def __init__(self, *args, **kwargs):
         """Initialize a DAOSVersion object."""
         super().__init__(*args, **kwargs)
+
         # Don't waste time starting servers and agents.
         self.setup_start_servers = False
         self.setup_start_agents = False
 
+        self.errors = []
+
+    def report_result(self):
+        """Helper printing test result"""
+        self.log.info("###### Test Result ######")
+        report_errors(test=self, errors=self.errors)
+        self.log.info("#########################")
+
+    def append_error(self, title, details):
+        """Helper adding an error to the list of errors
+
+        Args:
+            title (str): Error message title
+            details (list): List of string of the error details
+        """
+        msg = title
+        if details:
+            msg += "\n\t" + "\n\t".join(details)
+        self.errors.append(msg)
+
     def test_version(self):
         """Verify version number for dmg, daos, daos_server, and daos_agent against RPM.
 
         :avocado: tags=all,full_regression
         :avocado: tags=vm
         :avocado: tags=control,daos_cmd
-        :avocado: tags=version_number,test_version
+        :avocado: tags=DAOSVersion,test_version
         """
-        errors = []
-
         # Get RPM version.
         rpm_command = "rpm -qa|grep daos-server"
         output = run_pcmd(hosts=self.hostlist_servers, command=rpm_command)
-        self.log.info("RPM output = %s", output)
-        stdout = output[0]["stdout"][0]
-        self.log.info("RPM stdout = %s", stdout)
-        result = re.findall(r"daos-server-[tests-|tests_openmpi-]*([\d.]+)", stdout)
+        self.log.debug("RPM output = %s", output)
+        rc = output[0]["exit_status"]
+        stdout = output[0]["stdout"]
+        if rc != 0:
+            msg = "DAOS RPMs not properly installed: rc={}".format(rc)
+            self.append_error(msg, stdout)
+            self.report_result()
+        rpm_version = None
+        for rpm in stdout:
+            result = re.findall(r"daos-server-[tests-|tests_openmpi-]*([\d.]+)", rpm)
+            if result:
+                rpm_version = result[0]
+                break
         if not result:
-            errors.append("RPM version is not in the output! {}".format(output))
-        else:
-            rpm_version = result[0]
-            self.log.info("RPM version = %s", rpm_version)
+            msg = "RPM version could not be defined"
+            self.append_error(msg, stdout)
+            self.report_result()
+        self.log.info("RPM version = %s", rpm_version)
```

Comment:
I believe your code works, but I feel more comfortable moving this to line 73 because if you didn't know that `self.report_result()` on line 77 would fail the test, then you would think that this line will print `rpm_version` that is `None`.

Reply:
In fact, it was on purpose that it is after the `self.report_result()` to not print an informative message with an inconsistent RPM version.
However, if you think that it will be helpful to have this message, I have no issue to move it at line 73.

Reply:
I see, no problem. It's okay as it is.